### PR TITLE
cabal-install 3.4.0.0

### PR DESCRIPTION
**Pull request by chenrui333**
[`action-homebrew-bump-formula`](https://github.com/dawidd6/action-homebrew-bump-formula)

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by fishtreesugar** [comment]
FYI, cabal-install 3.4 changed its bootstrap script to [bootstrap.py](https://github.com/haskell/cabal/blob/3.4/bootstrap/bootstrap.py), we couldn't use the old way to bootstrap it anymore.

**Comment by carlocab** [comment]
@fishtreesugar would you like to open a new PR with the necessary updates to the formula? It might be a while before @chenrui333 finds the time to fix this.

File: Formula/cabal-install.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class CabalInstall < Formula
   desc "Command-line interface for Cabal and Hackage"
   homepage "https://www.haskell.org/cabal/"
-  url "https://hackage.haskell.org/package/cabal-install-3.2.0.0/cabal-install-3.2.0.0.tar.gz"
-  sha256 "a0555e895aaf17ca08453fde8b19af96725da8398e027aa43a49c1658a600cb0"
+  url "https://hackage.haskell.org/package/cabal-install-3.4.0.0/cabal-install-3.4.0.0.tar.gz"
+  sha256 "1980ef3fb30001ca8cf830c4cae1356f6065f4fea787c7786c7200754ba73e97"
   license "BSD-3-Clause"
-  revision 1
   head "https://github.com/haskell/cabal.git", branch: "3.2"
```
Comment (carlocab) [inline Formula/cabal-install.rb:7]:
```suggestion
  head "https://github.com/haskell/cabal.git", branch: "3.4"
```

**Comment by carlocab** [comment]
I don't think removing the patch is going to fix anything... The problem is that the tarball has no `bootstrap.sh`.

**Comment by chenrui333** [comment]
relates to https://github.com/haskell/cabal/issues/7289

**Comment by chenrui333** [comment]
yeah, it is indeed missing the bootstrap files for the artifact. (will revisit the PR later)

**Comment by carlocab** [comment]
Actually, it looks like we can bootstrap 3.4 with the older 3.2/

**Comment by fishtreesugar** [comment]
Sorry for the late reply.

> would you like to open a new PR with the necessary updates to the formula?

It seems a bit complicated to bootstrapping cabal-install without an environment with older cabal-install because the script mention above depending on some fixture generated by tools written by developers.

> Actually, it looks like we can bootstrap 3.4 with the older 3.2.

The [release.py](https://github.com/haskell/cabal/blob/master/release.py) seems to support it, but the script didn't bundle in the tarball too.

**Comment by carlocab** [comment]
I think I can work out how to bootstrap 3.4 with 3.2. However, I'm not sure which ghc to use. Would you know if I can build 3.4 with ghc 9.0.1 or 8.10.4?

**Comment by fishtreesugar** [comment]
According to the [dependencies](https://hackage.haskell.org/package/cabal-install-3.4.0.0) of cabal-install 3.4 and [bunded packages](https://downloads.haskell.org/ghc/9.0.1/docs/html/users_guide/9.0.1-notes.html#included-libraries) of GHC 9.0, the `base` package version of GHC 9.0 is `4.15.0.0`, but caball-install 3.4 require `>=4.8 && <4.15`.

I think we should use GHC 8.10.4.

**Comment by carlocab** [comment]
Ok. Giving this a try at #71832.

**Comment by fishtreesugar** [comment]
It looks like openbsd ports figured out  how to bootstrapping via bootstrap.py: https://github.com/blackgnezdo/ports/blob/ghc-8.10/devel/cabal/Makefile (haskell/cabal#7289)

**Comment by carlocab** [comment]
Won't work for us, because we need a `json` file that specifies the dependencies for `bootstrap.py`, like the one here:

https://github.com/blackgnezdo/ports/blob/83db945f59633c848a9ae28627cd8e378e316a09/devel/cabal/Makefile#L83

**Pull request closed without merging**